src/root.zig

```diff
@@ -673,36 +705,40 @@ pub const Unicode = struct {
 
     alg: Algorithm,
     opts: Options,
-    gcd: GenCatData,
-    cd: CaseData,
+    unicode_toolbox: UnicodeToolBox,
 
     pub fn init(
         allocator: std.mem.Allocator,
-        max_haystack: usize,
-        max_needle: usize,
         opts: Options,
     ) !Unicode {
-        // todo: these can likely be a static singleton that is shared amongst
-        // all instances of the Unicode fuzzy finder
-        var gcd = try GenCatData.init(allocator);
-        errdefer gcd.deinit();
+        const alg = try Algorithm.init(allocator);
+
+        const gcd = try GenCatData.init(allocator);
 
-        var cd = try CaseData.init(allocator);
-        errdefer cd.deinit();
+        var norm_data: *Normalize.NormData = undefined;
+        norm_data = try allocator.create(Normalize.NormData);
+        try Normalize.NormData.init(norm_data, allocator);
+        const norm = Normalize{ .norm_data = norm_data };
```

Comment:
Why does this need to be heap allocated? Also missing `errdefer` cleanups since `CaseData` can fail later.

Reply:
Yes, I forgot the `errdefer`. I will add them.

For the allocation, I follow the example provided in the documentation: https://codeberg.org/dude_the_builder/zg#normalization
Except one step, the line 719, that was to prevent `norm_data` to be `const`.
If you think it is a mistake, I don't find other way to do it.